doc/SONiC_Port_Mirroring_HLD.md

```diff
@@ -0,0 +1,468 @@
+
+### 3.5.5 Debug Commands
+Not applicable
+
+### 3.5.6 REST API Support
+
+- Please check all REST API from link @ https://<switch_ip>/ui link. 
+- This webserver provides user information about all the REST URLS, REST Data. Return codes. 
+- This webserver also provides interactive support to try REST queries.
+
+- Following REST SET and GET APIs will be supported
+
+The following show command display all the mirror sessions that are configured.
+```
+    # Get all mirror sessions
+    # curl -X GET "https://<switch_ip>/restconf/data/sonic-mirror-session:sonic-mirror-session" -H "accept: application/yang-data+json"
+
+    # Create SPAN session
+    # curl -X POST "https://<switch_ip>/restconf/data/sonic-mirror-session:sonic-mirror-session" -H "accept: application/yang-data+json" -H "Content-Type: application/yang-data+json" -d "{ \"sonic-mirror-session:MIRROR_SESSION\": { \"MIRROR_SESSION_LIST\": [ { \"name\": \"sess1\", \"dst_port\": \"Ethernet10\", \"src_port\": \"Ethernet8\", \"direction\": \"rx\" } ] }}"
+
+    # Delete all mirror sessions
+    # curl -X DELETE "https://<switch_ip>/restconf/data/sonic-mirror-session:sonic-mirror-session" -H "accept: application/yang-data+json"
+
+    # Delete specific mirror session
+    # curl -X DELETE "https://<switch_ip>/restconf/data/sonic-mirror-session:sonic-mirror-session/MIRROR_SESSION/MIRROR_SESSION_LIST=mirr3" -H "accept: application/yang-data+json"
+```
+
+### 3.5.7 GNMI Support
+
+- Following GNMI set and get commands will be supported.
+```
+    # Get all mirror sessions
+    # gnmi_get -xpath /sonic-mirror-session:sonic-mirror-session -target_addr 127.0.0.1:8080 -insecure
+
+    # Create SPAN session. mirror.json includes json payload same as rest-api above.
+    # gnmi_set -update /sonic-mirror-session:sonic-mirror-session/:@./mirror.json -target_addr 127.0.0.1:8080 -insecure
+
+    # Delete all mirror sessions
+    # gnmi_set -delete /sonic-mirror-session:sonic-mirror-session -target_addr 127.0.0.1:8080 -insecure
+
+    # Delete specific mirror session
+    # gnmi_set -delete /sonic-mirror-session:sonic-mirror-session/MIRROR_SESSION/MIRROR_SESSION_LIST[name=Mirror1] -target_addr 127.0.0.1:8080 -insecure
+```
+# 4 Flow Diagrams
+
+# 5 Error Handling
+
+- show mirror_session command will display any errors during session configuration and current status of session.
+- Internal processing errors within SwSS will be logged in syslog with ERROR level
+- SAI interaction errors will be logged in syslog
+
+# 6 Serviceability and Debug
+
+# 7 Warm Boot Support
+The mirroring configurations be retained across warmboot so that source traffic gets mirrored properly to destination port.
+
+# 8 Scalability
+
+Max mirror sessions supported are silicon specific. 
+
+###### Table 3: Scaling limits
+|Name                      |   Scaling value    |
+|--------------------------|--------------------|
+| Max mirror sessions      | silicon specific   |
+
```

Comment:
Do you have any scaling numbers?

Reply:
For Broadcom ASICs, max 4 sessions can be supported.